subprojects/base-services/src/main/java/org/gradle/internal/time/DefaultTimer.java

```diff
@@ -15,21 +15,18 @@
  */
 package org.gradle.internal.time;
 
+import java.util.concurrent.TimeUnit;
+
 class DefaultTimer implements Timer {
 
-    private final Clock clock;
+    private final TimeSource timeSource;
     private long startTime;
 
-    DefaultTimer(Clock clock) {
-        this.clock = clock;
+    DefaultTimer(TimeSource timeSource) {
```

Comment:
Why are we no longer using the clock here?

Reply:
We don't need its semantics, particularly when there's questions about “which clock?”.